### PR TITLE
Gif architecture

### PR DESCRIPTION
**Pull request by l-olson1214**
**Phabricator:** 
https://phabricator.wikimedia.org/T381331
https://phabricator.wikimedia.org/T380802 (accidentally included the work here!)

### Notes
* I had trouble getting the announcement to show up for me logged in because you need to reset it to see the announcement, but the code is in there for anyone who gets it the regular way

### Test Steps
1. In the proper region and language, with a logged in user with personalized slides
2. Load 6+ articles into history then background and foreground
3. Navigate to profile and then year in review
4. Compare images/gifs and copy to Figma, ensure correct.
5. Repeat with logged out user.
6. Check feature announcement while repeating with logged out user, ensure it is updated. 

TLDR:
With both logged in (with personalized slides) and logged out, run through the YIR and compare to Figma
Do the same with feature announcements

### Screenshots/Videos

https://github.com/user-attachments/assets/6c08a63a-3b4b-4425-905c-7109b60e4be8

https://github.com/user-attachments/assets/44a4ce92-c943-4ad9-942b-365b390e4992

simulator_screenshot_C4C27C5D-A498-4526-8789-748AE1C2DC06

### CODE REVIEW

File: Wikipedia/Code/YearInReviewCoordinator.swift [hunk 1]
```diff
@@ -78,22 +78,37 @@ final class YearInReviewCoordinator: NSObject, Coordinator {
...
+            personalizedWeekdayAccessibilityLabel: WMFLocalizedString("year-in-review-personalized-weekday", value: "A clock ticking, symbolizing the time spent by people reading Wikipedia.", comment: "Accessibility description for the personalized weekday slide."),
+            personalizedYourEditsViewsAccessibilityLabel: WMFLocalizedString("year-in-review-personalized-your-edits-views", value: "An illustration featuring a Wikipedia puzzle piece alongside a pen.", comment: "Accessibility description for the personalized 'Your Edits Views' slide."),
+            collectiveExploreAccessibilityLabel: WMFLocalizedString("year-in-review-collective-explore", value: "An animated illustration depicting the iconic Wikipedia globe encircled by scaffolding, symbolizing the collaborative effort of people worldwide in building the project. The scene is surrounded by a cityscape, emphasizing its connection to humanity and global culture.", comment: "Accessibility description for the collective explore slide."),
```
Comment (tonisevener) [inline Wikipedia/Code/YearInReviewCoordinator.swift:101]: (I could be totally reading this wrong, but please double-check) I think this `collectiveExploreAccessibilityLabel ` accessibility label is displayed with `english-slide-00`, and the description does not match that gif to me.

File: Wikipedia/Code/YearInReviewCoordinator.swift [hunk 2]
```diff
@@ -78,22 +78,37 @@ final class YearInReviewCoordinator: NSObject, Coordinator {
...
+            collectiveLanguagesAccessibilityLabel: WMFLocalizedString("year-in-review-collective-languages", value: "An animated illustration of a stone engraved with inscriptions representing various languages, symbolizing how Wikipedia collaboratively builds knowledge from diverse cultures and regions.", comment: "Accessibility description for the collective languages slide."),
+            collectiveArticleViewsAccessibilityLabel: WMFLocalizedString("year-in-review-collective-article-views", value: "An animated illustration depicting the iconic Wikipedia globe encircled by scaffolding, symbolizing the collaborative effort of people worldwide in building the project. The scene is surrounded by a cityscape, emphasizing its connection to humanity and global culture.", comment: "Accessibility description for the collective article views slide."),
```
Comment (tonisevener) [inline Wikipedia/Code/YearInReviewCoordinator.swift:103]: Also double-check this label - `collectiveArticleViewsAccessibilityLabel` is displayed alongside `english-slide-02`, which does not match the label.

File: Wikipedia/Code/YearInReviewCoordinator.swift [hunk 3]
```diff
@@ -78,22 +78,37 @@ final class YearInReviewCoordinator: NSObject, Coordinator {
...
+            collectiveAmountEditsAccessibilityLabel: WMFLocalizedString("year-in-review-collective-edits", value: " A graph showing the top 10 most edited Wikipedia language editions in 2024: English with over 31 million edits, German with 5,508,570 edits, French with 5,276,385 edits, Spanish with 4,786,205 edits, Russian with 3,303,066 edits, Italian with 3,200,398 edits, Japanese with 2,973,657 edits, Chinese with 2,505,032 edits, Polish with 1,383,808 edits, and Ukrainian with 1,376,980 edits. The total number of edits across all Wikipedia editions in 2024 is 81,987,181.", comment: "Accessibility description for the collective edits slide."),
+            collectiveEditsPerMinuteAccessibilityLabel: WMFLocalizedString("year-in-review-collective-edits-per-minute", value: "A clock ticking, symbolizing the time spent by people reading Wikipedia.", comment: "Accessibility description for the collective edits per minute slide."),
```
Comment (tonisevener) [inline Wikipedia/Code/YearInReviewCoordinator.swift:106]: Also check `collectiveEditsPerMinuteAccessibilityLabel` - it's displayed alongside `english-slide-05`. The label does not match the gif.

File: Wikipedia/Code/YearInReviewCoordinator.swift [hunk 4]
```diff
@@ -111,14 +126,17 @@ final class YearInReviewCoordinator: NSObject, Coordinator {
     // MARK: - Base Slide Strings
     
     var collectiveReadingBroughtUsTogetherSlideTitle: String {
```
Comment (tonisevener) [inline Wikipedia/Code/YearInReviewCoordinator.swift:128]: Since the value is no longer `Reading brought us together`, maybe we should rename the variable name too?

File: WMFComponents/Sources/WMFComponents/Components/Shared/Feature Announcement/WMFFeatureAnnouncementView.swift [hunk 5]
```diff
@@ -47,8 +47,23 @@ struct WMFFeatureAnnouncementView: View {
                                 .font(Font(WMFFont.for(.callout)))
                                 .foregroundColor(Color(appEnvironment.theme.text))
                         }
-
-                        if let image = viewModel.image {
+                        
+                        if let gifName = viewModel.gifName {
+                            ZStack {
+                                Image(gifName, bundle: .module)
```
Comment (tonisevener) [inline WMFComponents/Sources/WMFComponents/Components/Shared/Feature Announcement/WMFFeatureAnnouncementView.swift:53]: We may want to add an accessibility label on the image / gif in the feature announcement. It's reading out some strange stuff for me when I highlight it.

File: WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFYearInReviewScrollView.swift [hunk 6]
```diff
@@ -51,30 +45,16 @@ public struct WMFYearInReviewScrollView: View {
...
-                    if let overlayText = textOverlay {
-                        Text(overlayText)
-                            .font(Font(WMFFont.for(.xxlTitleBold)))
-                            .foregroundColor(.white)
-                    }
+                    WMFGIFfImageView(gifName)
+                        .aspectRatio(1.5, contentMode: .fit)
+                        .frame(maxWidth: .infinity)
                 }
+                .accessibilityLabel(altText)
```
Comment (tonisevener) [inline WMFComponents/Sources/WMFComponents/Components/Year in Review/WMFYearInReviewScrollView.swift:57]: It's reading "WMFComponents WMFComponents bundle, image" to me for all the gifs. Let me know if you can repro. So far with my tiny tweaks I haven't been able to fix it, you may have better luck.

**Review by tonisevener — COMMENTED** [review]
🎉 I love it - great work! I think we should address these comments, but I can merge now and you can fix them in your next PR so folks can start seeing it in builds.

File: Wikipedia/Code/YearInReviewCoordinator.swift (hunk 2 above)
Comment (tonisevener) [reply Wikipedia/Code/YearInReviewCoordinator.swift:103]: Update: I think this one needs to be `An animated illustration of a computer screen with a web browser open, actively navigating through a Wikipedia article.`

File: Wikipedia/Code/YearInReviewCoordinator.swift (hunk 3 above)
Comment (tonisevener) [reply Wikipedia/Code/YearInReviewCoordinator.swift:106]: Update: I think this one needs to be `An animated illustration showing bytes stacking on top of each other, symbolizing the continuous creation of free knowledge.`.
